syntax = "proto3";

package pixiv.v1;

// Authentication request
message ValidateAuthRequest {
  string refresh_token = 1;
}

message ValidateAuthResponse{
  bool success = 1;
}

// User information
message GetBookmarksRequest {
  string pixiv_user_id = 1;
  string refresh_token = 2;
  bool download_private = 3;
}

// Bookmark list response
message GetBookmarksResponse {
  repeated Bookmark bookmarks = 1;
}

// Individual bookmark
message Bookmark {
  string id = 1;
}

// Image request
message GetImageRequest {
  string artwork_id = 1;
  string refresh_token = 2;
  ImageQuality quality = 3;
  int32 page = 5;
}

// Image quality options
enum ImageQuality {
  THUMBNAIL = 0;
  MEDIUM = 1;
  ORIGINAL = 2;
}

// Image response with metadata
message GetImageResponse {
  bytes image_data = 1;
  string content_type = 2;
}

// Health check
message HealthCheckRequest {}
message HealthCheckResponse {
  string status = 1;
}

service PixivManager {
  // Get user bookmarks with pagination
  rpc GetBookmarks(GetBookmarksRequest) returns (GetBookmarksResponse);

  // Get image data for specific artwork
  rpc GetImage(GetImageRequest) returns (GetImageResponse);

  // Validate authentication token
  rpc ValidateAuth(ValidateAuthRequest) returns (ValidateAuthResponse);

  // Health check
  rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse);
}